### PR TITLE
Add base firmware for Silhouette

### PR DESCRIPTION
**Pull request by ToastyStoemp**
## Description

Add firmware for the Silhouette PCB by 4GT

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/wolf/silhouette/config.h
Comment (waffle87) [inline keyboards/wolf/silhouette/config.h:1]: This file should be removed.

File: keyboards/wolf/silhouette/rules.mk [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# This file intentionally left blank
+WS2812_DRIVER = vendor
+
```
Comment (waffle87) [inline keyboards/wolf/silhouette/rules.mk:3]:
```suggestion
WS2812_DRIVER = vendor
```

File: keyboards/wolf/silhouette/readme.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+The PCB features:
+* QMK & VIA compatibility
+
+---
+
+* Keyboard Maintainer: [ToastyStoemp](https://github.com/ToastyStoemp)
+
+Make example for this keyboard (after setting up your build environment):
+
+	make wolf/silhouette:default
```
Comment (waffle87) [inline keyboards/wolf/silhouette/readme.md:14]: This portion should follow the [template](https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/readme.md?plain=1).
ie. the hardware availability and whatnot.

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/wolf/silhouette/readme.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+* Hardware Availability: https://beacons.ai/chamchi
+
+The PCB features:
+* QMK & VIA compatibility
+
+---
+
+Make example for this keyboard (after setting up your build environment):
+
+	make wolf/silhouette:default
```
Comment (waffle87) [inline keyboards/wolf/silhouette/readme.md:18]:
```suggestion
    make wolf/silhouette:default

Flashing example for this keyboard:

    make wolf/silhouette:default:flash

```

**Review by waffle87 — APPROVED** [review]

File: keyboards/wolf/silhouette/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+    [0] = LAYOUT(
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_DEL,    KC_INS,  KC_HOME, KC_PGUP,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,            KC_DEL,  KC_END,  KC_PGDN,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_NUHS, KC_ENT,         
+        KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),                       KC_UP,
```
Comment (zvecr) [inline keyboards/wolf/silhouette/keymaps/default/keymap.c:24]: `MO(1)` here goes out of bounds of the keymap array

File: keyboards/wolf/silhouette/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    
+    [0] = LAYOUT(
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_DEL,    KC_INS,  KC_HOME, KC_PGUP,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,            KC_DEL,  KC_END,  KC_PGDN,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_NUHS, KC_ENT,         
+        KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),                       KC_UP,
```
Comment (zvecr) [inline keyboards/wolf/silhouette/keymaps/via/keymap.c:24]: Same for this one too

File: keyboards/wolf/silhouette/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
```
Comment (zvecr) [inline keyboards/wolf/silhouette/keymaps/default/keymap.c:13]: Shouldnt mix copyright header style
```suggestion
// Copyright 2023 contact@vwolf.be
```

File: keyboards/wolf/silhouette/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
```
Comment (zvecr) [inline keyboards/wolf/silhouette/keymaps/via/keymap.c:13]:
```suggestion
// Copyright 2023 contact@vwolf.be
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/wolf/silhouette/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,24 @@
+/*
```
Comment (fauxpark) [inline keyboards/wolf/silhouette/keymaps/default/keymap.c:1]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/wolf/silhouette/readme.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+* Keyboard Maintainer: [Wolf](https://github.com/ToastyStoemp)
+* Hardware Supported: Silhouette PCB
+* Hardware Availability: https://beacons.ai/chamchi
+
+The PCB features:
+* QMK & VIA compatibility
+
+---
+
```
Comment (fauxpark) [inline keyboards/wolf/silhouette/readme.md:15]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/wolf/silhouette/readme.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
+# Silhouette
+
+![Silhouette](https://i.imgur.com/x5Wo7t1.jpeg)
```
Comment (drashna) [inline keyboards/wolf/silhouette/readme.md:3]:
```suggestion
![Silhouette](https://i.imgur.com/x5Wo7t1h.jpeg)
```

File: keyboards/wolf/silhouette/rules.mk [hunk 11]
```diff
@@ -0,0 +1 @@
+WS2812_DRIVER = vendor
```
Comment (fauxpark) [inline keyboards/wolf/silhouette/rules.mk:1]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/wolf/silhouette/rules.mk (hunk 11 above)
Comment (ToastyStoemp) [reply keyboards/wolf/silhouette/rules.mk:1]: Is vendor now the default?

Comment (fauxpark) [reply keyboards/wolf/silhouette/rules.mk:1]: No, I don't see where the WS2812 driver is configured. You haven't defined a pin or enabled RGBlight/RGB Matrix.

Comment (ToastyStoemp) [reply keyboards/wolf/silhouette/rules.mk:1]: oh that's fair, sorry about that

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**